gcloud/datastore/test_key.py

```diff
@@ -31,41 +32,53 @@ def tearDown(self):
         _implicit_environ.DATASET_ID = self._replaced_dataset_id
 
     def _getTargetClass(self):
-        from gcloud.datastore import _implicit_environ
-        from gcloud.datastore.dataset import Dataset
         from gcloud.datastore.key import Key
-
-        _implicit_environ.DATASET = Dataset(self._DEFAULT_DATASET)
         return Key
 
     def _makeOne(self, *args, **kwargs):
         return self._getTargetClass()(*args, **kwargs)
 
+    def _monkeyDatasetID(self, dataset_id=_DEFAULT_DATASET):
```

Comment:
They make it explicit which tests rely on the implicit dataset ID (`setUp` and `tearDown` hide that).